src/cleanup.ts

```diff
@@ -82,12 +94,11 @@ async function collectAndPublishBuildInfoIfNeeded() {
 
     // The flow here is to collect Git information before publishing the build info.
     // We allow this step to fail, and we don't want to fail the entire build publish if they do.
-
     try {
         core.startGroup('Collect the Git information');
         await Utils.runCli(['rt', 'build-add-git'], { cwd: workingDirectory });
     } catch (error) {
-        core.warning('failed while attempting to collect Git information: ' + error);
+        core.warning('Failed while attempting to collect Git information: ' + error);
     } finally {
         core.endGroup();
     }
```

Comment:
Let's wrap the group below with try-finally to ensure group closing after failure
```suggestion
    }
    
    // Publish the build info to Artifactory
    try {
        core.startGroup('Publish the build info to JFrog Artifactory');
        await Utils.runCli(['rt', 'build-publish'], { cwd: workingDirectory });
    } finally {
        core.endGroup();
    }
```